api/configuration/v1alpha1/konnect_controlplaneref_types.go

```diff
@@ -1,39 +1,41 @@
 package v1alpha1
 
 const (
-	ControlPlaneRefKonnectID            = "konnectID"
+	// ControlPlaneRefKonnectID is the type for the KonnectID ControlPlaneRef.
+	// It is used to reference a Konnect Control Plane entity by its ID on the Konnect platform.
+	ControlPlaneRefKonnectID = "konnectID"
+	// ControlPlaneRefKonnectNamespacedRef is the type for the KonnectNamespacedRef ControlPlaneRef.
+	// It is used to reference a Konnect Control Plane entity inside the cluster
+	// using a namespaced reference.
 	ControlPlaneRefKonnectNamespacedRef = "konnectNamespacedRef"
-	ControlPlaneRefKIC                  = "kic"
 )
 
+// ControlPlaneRef is the schema for the ControlPlaneRef type.
+// It is used to reference a Control Plane entity.
+// +kubebuilder:validation:XValidation:rule="self.type == 'konnectNamespacedRef' ? has(self.konnectNamespacedRef) : true", message="when type is konnectNamespacedRef, konnectNamespacedRef must be set"
```

Comment:
I think we should add the same kind of validation for `KonnectID`. WDYT?

Reply:
8eafac17c1c670ee370f09b0001c79e507717f86

PTAL